support/stdio.c

```diff
@@ -202,7 +202,7 @@ Mono_Posix_Stdlib_fgetc (void* stream)
 	return fgetc (stream);
 }
 
-char*
+void*
```

Comment:
Why `void*`?  `fgets` returns a `char*`

Reply:
The return type has to match the prototype in `map.h`, and the prototype in `map.h` is autogenerated from `Stdlib.sys_fgets` which returns `IntPtr` (which is mapped to `void*`): https://github.com/mono/mono/blob/ad2132e611a7e8b7c7dcaed1c3ecc23d40dc17de/mcs/class/Mono.Posix/Mono.Unix.Native/Stdlib.cs#L888

Reply:
We only check the pointer against null in Stdlib.cs so the type isn't really relevant :)